contracts/src/ethereum/TokenBridge.sol

```diff
@@ -139,4 +139,28 @@ contract TokenBridge is
         require(l1Token.balanceOf(msg.sender) - amount <= l1Token.balanceOf(msg.sender), "UNDERFLOW");
         l1Token.transfer(recipient, amount);
     }
+
+     function consumeBridgeRewardMessage(address l1Token, address recipient, uint256 amount) internal {
+        emit LogBridgeReward(l1Token, recipient, amount);
+
+        uint256[] memory payload = new uint256[](5);
+        payload[0] = BRIDGE_REWARD_MESSAGE;
+        payload[1] = uint256(l1Token);
+        payload[1] = uint256(recipient);
+        payload[2] = amount & (UINT256_PART_SIZE - 1);
+        payload[3] = amount >> UINT256_PART_SIZE_BITS;
+
+        messagingContract.consumeMessageFromL2(l2TokenBridge, payload);
+    }
+
+    function receiveRewards(address l1Token, address recipient, uint256 amount) isApprovedToken(l1Token) external {
+        consumeBridgeRewardMessage(l1Token, recipient, amount);
+        require(recipient != address(0x0), "INVALID_RECIPIENT");
```

Comment:
Shouldn't we switch the lines 157 and 158?

Reply:
I think we should consume any messages from l2 first incase their somehow was a message with the recipient set to 0.

Reply:
Indeed, I agree with this logic